components/zcash/rs/lib.rs

```diff
@@ -2,27 +2,92 @@
 // This Source Code Form is subject to the terms of the Mozilla Public
 // License, v. 2.0. If a copy of the MPL was not distributed with this file,
 // You can obtain one at https://mozilla.org/MPL/2.0/.
+
+mod librustzcash;
+
 use std::fmt;
 
 use orchard::{
+    builder:: {
+        BuildError as OrchardBuildError,
+        InProgress,
+        Unproven,
+        Unauthorized
+    },
+    bundle::Bundle,
     zip32::ChildIndex as OrchardChildIndex,
     keys::Scope as OrchardScope,
     keys::FullViewingKey as OrchardFVK,
     zip32::Error as Zip32Error,
-    zip32::ExtendedSpendingKey};
+    zip32::ExtendedSpendingKey,
+    tree::MerkleHashOrchard
+};
+
+use crate::librustzcash::amount::Amount;
+
+use ffi::OrchardOutput;
+
+use rand_core::OsRng;
+use rand_core::{RngCore, Error as OtherError, impls};
+use rand_core::CryptoRng;
 
 use brave_wallet_cxx::{
   impl_result,
   impl_error
 };
 
+#[derive(Clone)]
+pub(crate) struct MockRng(u64);
+
+impl CryptoRng for MockRng {}
+
+impl RngCore for MockRng {
+    fn next_u32(&mut self) -> u32 {
+        self.next_u64() as u32
+    }
+
+    fn next_u64(&mut self) -> u64 {
+        self.0 += 1;
```

Comment:
This is used only in tests, do you think we really need to use complicated randomness here?

Reply:
It's only used in test code, but it's available in all builds. I'm concerned about someone else using it accidentally, sometime in the future, where they need something else.

rand_chacha is a `SeedableRng + CryptoRng` we already have in tree, so using it is also just less code.

Reply:
We have discussed with @kdenhartog. He is ok with keeping mock rng since it's call is protected by CHECK_IS_TESTS. Otherwise i'd need to recapture testvectors.